File: lib/lang/bs.js


function join_with_shared_prefix(a, b, joiner) {
  let i = 0;

  while(i !== a.length &&
        i !== b.length &&
        a.charCodeAt(i) === b.charCodeAt(i))
    ++i;

  while(i && a.charCodeAt(i - 1) !== 32)
    --i;

  return a + joiner + b.slice(i);
}

module.exports = {
  "clear": "vedro",
  "no-precipitation": "nema padavina",
  "mixed-precipitation": "različite padavine",
  "possible-very-light-precipitation": "moguće su slabe padavine",
  "very-light-precipitation": "slabe padavine",
  "possible-light-precipitation": "moguće su slabe padavine",
  "light-precipitation": "slabe padavine",
  "medium-precipitation": "padavine",
  "heavy-precipitation": "jake padavine",
  "possible-very-light-rain": "moguće je sitna kiša",
  "very-light-rain": "sitna kiša",
  "possible-light-rain": "moguće je sitna kiša",
  "light-rain": "sitna kiša",
  "medium-rain": "kiša",
  "heavy-rain": "jaka kiša",
  "possible-very-light-sleet": "moguće je slaba susnježica",
  "very-light-sleet": "slaba susnježica",
  "possible-light-sleet": "moguće je slaba susnježica",
  "light-sleet": "slaba susnježica",
  "medium-sleet": "susnježica",
  "heavy-sleet": "jaka susnježica",
  "possible-very-light-snow": "moguće je sitan snijeg",
  "very-light-snow": "sitan snijeg",
  "possible-light-snow": "moguće je sitan snijeg",
  "light-snow": "sitan snijeg",
  "medium-snow": "snijeg",
  "heavy-snow": "jak snijeg",
  "possible-thunderstorm": "moguće je grmljavine",
  "thunderstorm": "grmljavine",
  "possible-medium-precipitation": "moguće su padavine",
  "possible-heavy-precipitation": "moguće su jake padavine",
  "possible-medium-rain": "moguće je kiša",
  "possible-heavy-rain": "moguće je jaka kiša",
  "possible-medium-sleet": "moguće je susnježica",
  "possible-heavy-sleet": "moguće je jaka susnježica",
  "possible-medium-snow": "moguće je snijeg",
  "possible-heavy-snow": "moguće je jaka snijeg",
  "possible-freezing-drizzle": "moguće je ledena kiša",
  "freezing-drizzle": "ledena kiša",
  "possible-light-freezing-rain": "moguće je sitan ledena kiša",
  "light-freezing-rain": "sitan ledena kiša",
  "possible-freezing-rain": "moguće je ledena kiša",
  "freezing-rain": "ledena kiša",
  "possible-heavy-freezing-rain": "moguće je jaka ledena kiša",
  "heavy-freezing-rain": "jaka ledena kiša",
  "possible-hail": "moguća grmljavina sa gradom",
  "hail": "grmljavina sa gradom",
  "light-wind": "vjetrovito",
  "medium-wind": "vjetrovito",
  "heavy-wind": "opasno vjetrovito",
  "low-humidity": "suho",
  "high-humidity": "vlažno",
  "fog": "maglovito",
  "very-light-clouds": "pretežno vedro",
  "light-clouds": "djelom oblačno",
  "medium-clouds": "pretežno oblačno",
  "heavy-clouds": "oblačno",
  "today-morning": "ovo jutro",
  "later-today-morning": "kasnije ovog jutra",
  "today-afternoon": "ove podne",
  "later-today-afternoon": "kasnije ove podne",
  "today-evening": "večeras",
  "later-today-evening": "kasnije večeras",
  "today-night": "u noć",
  "later-today-night": "kasnije u noć",
  "tomorrow-morning": "sutra ujutro",
  "tomorrow-afternoon": "sutra popodne",
  "tomorrow-evening": "sutra uveče",
  "tomorrow-night": "sutra u noć",
  "morning": "ujutro",
  "afternoon": "popodne",
  "evening": "uveče",
  "night": "u noć",
  "today": "danas",
  "tomorrow": "sutra",
  "sunday": "u nedjelju",
  "monday": "u ponedjeljak",
  "tuesday": "u utorak",
  "wednesday": "u srijeda",
  "thursday": "u četvrtak",
  "friday": "u petak",
  "saturday": "u subotu",
  "next-sunday": "u nedjelju", // FIXME
  "next-monday": "u ponedjeljak", // FIXME
  "next-tuesday": "u utorak", // FIXME
  "next-wednesday": "u srijeda", // FIXME
  "next-thursday": "u četvrtak", // FIXME
  "next-friday": "u petak", // FIXME
  "next-saturday": "u subotu", // FIXME
  "minutes": "$1 minuta",
  "fahrenheit": "$1\u00B0F",
  "celsius": "$1\u00B0C",
  "inches": "$1 inča",
  "centimeters": "$1 centimetra",
  "less-than": "ispod $1",
  "and": function(a, b) {
    return join_with_shared_prefix(
      a,
      b,
      a.indexOf(",") !== -1 ? ", i " : " i ",
    );
  },
  "through": function(a, b) {
    return join_with_shared_prefix(a, b, " do ");
  },
  "with": "$1, sa $2",
  "range": "$1\u2013$2",
  "parenthetical": "$1 ($2)",
  "for-hour": "$1 za ovaj sat",
  "starting-in": "$1 počinje za $2",
  "stopping-in": "$1 prekida za $2",
  "starting-then-stopping-later": "$1 počinje za $2, ondak prekida za $3",
  "stopping-then-starting-later": "$1 prekida za $2, ondak počinje za $3",
  "for-day": "$1 tokom cijelog dana",
  "starting": "$1 počinje $2",
  "until": "$1 do $2",
  "until-starting-again": "$1 do $2, i opet poćinje $3",
  "starting-continuing-until": "$1 poćinje $2, ostaje do $3",
  "during": "$1 $2",
  "for-week": "$1 tokom sedmice",
  "over-weekend": "$1 tokom vikenda",
  "temperatures-peaking": "temperaturom najviše do $1 $2",
  "temperatures-rising": "temperaturom raste do $1 $2",
  "temperatures-valleying": "temperaturom najniže do $1 $2",
  "temperatures-falling": "temperaturom pada do $1 $2",
  // Capitalize the first letter of every word, except if that word is
  // "and". (This is a very crude bastardization of proper English titling
  // rules, but it is adequate for the purposes of this module.)
  "title": function(str) {
    return str.charAt(0).toUpperCase() + str.slice(1);
  },
  /* Capitalize the first word of the sentence and end with a period. */
  "sentence": function(str) {
    /* Capitalize. */
    str = str.charAt(0).toUpperCase() + str.slice(1);

    /* Add a period if there isn't already one. */
    if(str.charAt(str.length - 1) !== ".")
      str += ".";

    return str;
  },
};

File: test_cases/bs.py
# This would be one of your translations, as a Python dictionary
cases = {
    "Vedro": ["title", "clear"],
    "Moguće su slabe padavine": ["title", "possible-very-light-precipitation"],
    "Slabe padavine": ["title", "very-light-precipitation"],
    "Moguće su slabe padavine": ["title", "possible-light-precipitation"],
    "Slabe padavine": ["title", "light-precipitation"],
    "Padavine": ["title", "medium-precipitation"],
    "Jake padavine": ["title", "heavy-precipitation"],
    "Moguće je sitna kiša": ["title", "possible-very-light-rain"],
    "Sitna kiša": ["title", "very-light-rain"],
    "Moguće je sitna kiša": ["title", "possible-light-rain"],
    "Sitna kiša": ["title", "light-rain"],
    "Kiša": ["title", "medium-rain"],
    "Jaka kiša": ["title", "heavy-rain"],
    "Moguće je slaba susnježica": ["title", "possible-very-light-sleet"],
    "Slaba susnježica": ["title", "very-light-sleet"],
    "Moguće je slaba susnježica": ["title", "possible-light-sleet"],
    "Slaba susnježica": ["title", "light-sleet"],
    "Susnježica": ["title", "medium-sleet"],
    "Jaka susnježica": ["title", "heavy-sleet"],
    "Moguće je sitan snijeg": ["title", "possible-very-light-snow"],
    "Sitan snijeg": ["title", "very-light-snow"],
    "Moguće je sitan snijeg": ["title", "possible-light-snow"],
    "Sitan snijeg": ["title", "light-snow"],
    "Snijeg": ["title", "medium-snow"],
    "Jak snijeg": ["title", "heavy-snow"],
    "Vjetrovito": ["title", "medium-wind"],
    "Opasno vjetrovito": ["title", "heavy-wind"],
    "Maglovito": ["title", "fog"],
    "Pretežno vedro": ["title", "very-light-clouds"],
    "Djelom oblačno": ["title", "light-clouds"],
    "Pretežno oblačno": ["title", "medium-clouds"],
    "Oblačno": ["title", "heavy-clouds"],
    "Suho i vjetrovito": ["title", ["and", "low-humidity", "light-wind"]],
    "Sitna kiša i opasno vjetrovito": [
        "title",
        ["and", "very-light-rain", "heavy-wind"],
    ],
    "Vlažno i djelom oblačno": ["title", ["and", "high-humidity", "light-clouds"]],
    "Vedro za ovaj sat.": ["sentence", ["for-hour", "clear"]],
    "Sitan snijeg počinje za 35 minuta.": [
        "sentence",
        ["starting-in", "very-light-snow", ["minutes", 35]],
    ],
    "Sitna kiša prekida za 15 minuta.": [
        "sentence",
        ["stopping-in", "light-rain", ["minutes", 15]],
    ],
    "Jaka susnježica počinje za 20 minuta, ondak prekida za 30 minuta.": [
        "sentence",
        [
            "starting-then-stopping-later",
            "heavy-sleet",
            ["minutes", 20],
            ["minutes", 30],
        ],
    ],
    "Kiša prekida za 25 minuta, ondak počinje za 8 minuta.": [
        "sentence",
        [
            "stopping-then-starting-later",
            "medium-rain",
            ["minutes", 25],
            ["minutes", 8],
        ],
    ],
    "Pretežno oblačno tokom cijelog dana.": ["sentence", ["for-day", "medium-clouds"]],
    "Slaba susnježica počinje ujutro.": [
        "sentence",
        ["starting", "very-light-sleet", "morning"],
    ],
    "Vjetrovito do u noć.": ["sentence", ["until", "medium-wind", "today-night"]],
    "Jake padavine do popodne.": [
        "sentence",
        ["until", "heavy-precipitation", "afternoon"],
    ],
    "Vjetrovito popodne.": ["sentence", ["during", "light-wind", "afternoon"]],
    "Snijeg kasnije večeras i sutra ujutro.": [
        "sentence",
        ["during", "medium-snow", ["and", "later-today-evening", "tomorrow-morning"]],
    ],
    "Jaka kiša do kasnije ovog jutra, i opet poćinje večeras.": [
        "sentence",
        ["until-starting-again", "heavy-rain", "later-today-morning", "today-evening"],
    ],
    "Oblačno poćinje uveče, ostaje do u noć.": [
        "sentence",
        ["starting-continuing-until", "heavy-clouds", "evening", "night"],
    ],
    "Slaba susnježica kasnije ove podne i maglovito sutra ujutro.": [
        "sentence",
        [
            "and",
            ["during", "light-sleet", "later-today-afternoon"],
            ["during", "fog", "tomorrow-morning"],
        ],
    ],
    "Opasno vjetrovito poćinje ovo jutro, ostaje do ove podne, i susnježica sutra ujutro.": [
        "sentence",
        [
            "and",
            [
                "starting-continuing-until",
                "heavy-wind",
                "today-morning",
                "today-afternoon",
            ],
            ["during", "medium-sleet", "tomorrow-morning"],
        ],
    ],
    "Oblačno počinje kasnije u noć i jak snijeg sutra popodne.": [
        "sentence",
        [
            "and",
            ["starting", "heavy-clouds", "later-today-night"],
            ["during", "heavy-snow", "tomorrow-afternoon"],
        ],
    ],
    "Suho u noć i slabe padavine poćinje sutra uveče, ostaje do sutra u noć.": [
        "sentence",
        [
            "and",
            ["during", "low-humidity", "today-night"],
            [
                "starting-continuing-until",
                "light-precipitation",
                "tomorrow-evening",
                "tomorrow-night",
            ],
        ],
    ],
    "Snijeg (5 inča) u noć.": [
        "sentence",
        ["during", ["parenthetical", "medium-snow", ["inches", 5]], "night"],
    ],
    "Sitan snijeg (2 centimetra) kasnije ovog jutra.": [
        "sentence",
        [
            "during",
            ["parenthetical", "light-snow", ["centimeters", 2]],
            "later-today-morning",
        ],
    ],
    "Jak snijeg (8\u201312 inča) tokom cijelog dana.": [
        "sentence",
        ["for-day", ["parenthetical", "heavy-snow", ["inches", ["range", 8, 12]]]],
    ],
    "Snijeg (ispod 1 centimetra) popodne.": [
        "sentence",
        [
            "during",
            ["parenthetical", "medium-snow", ["less-than", ["centimeters", 1]]],
            "afternoon",
        ],
    ],
    "Nema padavina tokom sedmice, sa temperaturom najviše do 85°F sutra.": [
        "sentence",
        [
            "with",
            ["for-week", "no-precipitation"],
            ["temperatures-peaking", ["fahrenheit", 85], "tomorrow"],
        ],
    ],
    "Različite padavine tokom vikenda, sa temperaturom raste do 32\u00b0C u četvrtak.": [
        "sentence",
        [
            "with",
            ["over-weekend", "mixed-precipitation"],
            ["temperatures-rising", ["celsius", 32], "thursday"],
        ],
    ],
    "Sitna kiša u ponedjeljak, sa temperaturom najniže do 15°F u petak.": [
        "sentence",
        [
            "with",
            ["during", "very-light-rain", "monday"],
            ["temperatures-valleying", ["fahrenheit", 15], "friday"],
        ],
    ],
    "Sitan snijeg u utorak i srijeda, sa temperaturom pada do 0\u00b0C u nedjelju.": [
        "sentence",
        [
            "with",
            ["during", "light-snow", ["and", "tuesday", "wednesday"]],
            ["temperatures-falling", ["celsius", 0], "sunday"],
        ],
    ],
    "Padavine danas do u subotu, sa temperaturom najviše do 100\u00b0F u ponedjeljak.": [
        "sentence",
        [
            "with",
            ["during", "medium-precipitation", ["through", "today", "saturday"]],
            ["temperatures-peaking", ["fahrenheit", 100], "monday"],
        ],
    ],
    "Dim": ['title', 'smoke'],
    "Dim tokom cijelog dana.": ['sentence', ['for-day', 'smoke']],
    "Dim ujutro.": ['sentence', ['during', 'smoke', 'morning']],
    "Dim do večeras.": ['sentence', ['until', 'smoke', 'today-evening']],
    "Dim i djelom oblačno": ['title', ['and', 'smoke', 'light-clouds']],
    "Izmaglica": ['title', 'haze'],
    "Izmaglica tokom cijelog dana.": ['sentence', ['for-day', 'haze']],
    "Izmaglica popodne.": ['sentence', ['during', 'haze', 'afternoon']],
    "Izmaglica i vlažno": ['title', ['and', 'haze', 'high-humidity']],
    "Magla": ['title', 'mist'],
    "Magla tokom cijelog dana.": ['sentence', ['for-day', 'mist']],
    "Magla u noć.": ['sentence', ['during', 'mist', 'night']],
    "Magla i oblačno": ['title', ['and', 'mist', 'heavy-clouds']],
    "Grmljavine za ovaj sat.": ["sentence", ["for-hour", "thunderstorm"]],
    "Grmljavina sa gradom počinje za 5 minuta, ondak prekida za 45 minuta.": [
        "sentence",
        [
            "starting-then-stopping-later",
            "hail",
            ["minutes", 5],
            ["minutes", 45],
        ],
    ],
    "Jaka ledena kiša prekida za 10 minuta, ondak počinje za 32 minuta.": [
        "sentence",
        [
            "stopping-then-starting-later",
            "heavy-freezing-rain",
            ["minutes", 10],
            ["minutes", 32],
        ],
    ],
    "Ledena kiša počinje za 49 minuta.": [
        "sentence",
        ["starting-in", "very-light-freezing-rain", ["minutes", 49]],
    ],
}
